### ISSUE TITLE
Small classes an small methods

### ISSUE THREAD

**Issue by szepeviktor** [sugarcalendar/icalendar-rrule-sequencer#3]
Hello @JJJ! You are raising a domesticated monster here :)

Could it be split up to small classes/traits and short methods?

**Comment by JJJ** [comment]
Yes, this certainly could and should be broken up into easier to understand parts.

Everything is pretty intertwined, and really it is one big bunch of dependencies, so if you have ideas I'm all ears 😅

**Comment by szepeviktor** [comment]
Sorry, I'm not a user.
A bunch of empty sheets and a pencil could help ✏️
